pkg/daemon/ceph/osd/daemon.go

```diff
@@ -84,10 +86,42 @@ func StartOSD(context *clusterd.Context, osdType, osdID, osdUUID, lvPath string,
 		if err := releaseLVMDevice(context, volumeGroupName); err != nil {
 			return fmt.Errorf("failed to release device from lvm. %+v", err)
 		}
+
 	}
+
 	return nil
 }
 
+func handleTerminate(context *clusterd.Context, lvPath, volumeGroupName string) error {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	for {
+		select {
+		case <-sigc:
+			logger.Infof("shutdown signal received, exiting...")
+
+			pid, err := context.Executor.ExecuteCommandWithOutput(false, "", "fuser", "-a", lvPath)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve process ID for - %s. Error %+v", lvPath, err)
+			}
+
+			logger.Infof("process ID for logical volume  %s : %s", lvPath, pid)
+			// shut down the osd-ceph process so that lvm release does not show device in use error.
+			if pid != "" {
+				if err := context.Executor.ExecuteCommand(false, "", "kill", pid); err != nil {
+					return fmt.Errorf("failed to delete ceph-osd process. %+v", err)
+				}
+			}
```

Comment:
You must wait here for that PID to be gone, give the time to the ceph-osd to terminate properly.
Or add a retry loop on `releaseLVMDevice`.